libsolidity/codegen/CompilerContext.cpp

```diff
@@ -127,6 +127,8 @@ void CompilerContext::addVariable(VariableDeclaration const& _declaration,
 								  unsigned _offsetToCurrent)
 {
 	solAssert(m_asm->deposit() >= 0 && unsigned(m_asm->deposit()) >= _offsetToCurrent, "");
+	unsigned sizeOnStack = _declaration.annotation().type->sizeOnStack();
+	solAssert(sizeOnStack == 1 || sizeOnStack == 2, "");
```

Comment:
Perhaps add a comment: variables should not have other stack sizes, but that might change when new types are introduced.